src/functions/authorizer.ts

```diff
@@ -29,69 +17,27 @@ const accessToHttpVerbs: AccessHttpVerbMap = {
  * @returns - Promise<APIGatewayAuthorizerResult>
  */
 export const authorizer = async (event: APIGatewayTokenAuthorizerEvent, context: Context): Promise<APIGatewayAuthorizerResult> => {
+  const logEvent: ILogEvent = {};
   try {
     initialiseLogEvent(event);
-    // fail-fast if config is missing or invalid
-    const config: AuthorizerConfig = await configuration();
-
-    const jwt: any = getValidJwt(event.authorizationToken);
-
-    const validRoles: Role[] = getValidRoles(jwt);
-
-    if (!validRoles || validRoles.length === 0) {
-      reportNoValidRoles(jwt, event, context);
-      writeLogMessage(logEvent, JWT_MESSAGE.INVALID_ROLES);
-      return unauthorisedPolicy();
-    }
-    // by this point we know authorizationToken meets formatting requirements
-    // remove 'Bearer ' when verifying signature
-    await checkSignature(event.authorizationToken.substring(7), jwt);
+    const jwt: any = await getValidJwt(event.authorizationToken, logEvent);
 
-    let statements: Statement[] = [];
+    const policy = generateRolePolicy(jwt, logEvent) ?? (await generateFunctionalPolicy(jwt, logEvent));
 
```

Comment:
Nitpick: this either/or `??` could be extracted to a `generatePolicy()`...  the arguments and return type are the same so it doesn't matter to the caller if they get a Role or Functional policy back (though putting a Factory in front of a Factory might be more confusing!)

Reply:
At first glance I'm not sure what benefit that gives? This is a one liner to show priority...could you help me see how abstracting this away helps?

Reply:
it's a nitpick, so I'm happy with the code as-is - resolving.